tests/framework/layer/test_layer.cpp

```diff
@@ -245,6 +245,56 @@ VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL get_instance_func(VkInstance instance,
 // Exported functions
 extern "C" {
 #if TEST_LAYER_EXPORT_ENUMERATE_FUNCTIONS
+
+// Pre-instance handling functions
+
+FRAMEWORK_EXPORT VKAPI_ATTR VkResult VKAPI_CALL test_preinst_vkEnumerateInstanceLayerProperties(
+    const VkEnumerateInstanceLayerPropertiesChain* pChain, uint32_t* pPropertyCount, VkLayerProperties* pProperties) {
+    VkResult res = pChain->pfnNextLayer(pChain->pNextLink, pPropertyCount, pProperties);
+    if (nullptr == pProperties) {
+        *pPropertyCount = 43;
+    } else {
+        uint32_t count = 43;
+        if (*pPropertyCount < 43) {
+            count = *pPropertyCount;
+            res = VK_INCOMPLETE;
+        }
+        for (uint32_t i = 0; i < count; ++i) {
+            snprintf(pProperties[i].layerName, VK_MAX_EXTENSION_NAME_SIZE, "%02d_layer", count);
+            pProperties[i].specVersion = count;
+            pProperties[i].implementationVersion = 0xABCD0000 + count;
+        }
+    }
+    return res;
+}
+
+FRAMEWORK_EXPORT VKAPI_ATTR VkResult VKAPI_CALL test_preinst_vkEnumerateInstanceExtensionProperties(
+    const VkEnumerateInstanceExtensionPropertiesChain* pChain, const char* pLayerName, uint32_t* pPropertyCount,
+    VkExtensionProperties* pProperties) {
+    VkResult res = pChain->pfnNextLayer(pChain->pNextLink, pLayerName, pPropertyCount, pProperties);
+    if (nullptr == pProperties) {
+        *pPropertyCount = 52;
+    } else {
+        uint32_t count = 52;
+        if (*pPropertyCount < 52) {
+            count = *pPropertyCount;
+            res = VK_INCOMPLETE;
+        }
+        for (uint32_t i = 0; i < count; ++i) {
+            snprintf(pProperties[i].extensionName, VK_MAX_EXTENSION_NAME_SIZE, "%02d_ext", count);
+            pProperties[i].specVersion = count;
+        }
+    }
+    return res;
+}
+
+FRAMEWORK_EXPORT VKAPI_ATTR VkResult VKAPI_CALL
+test_preinst_vkEnumerateInstanceVersion(const VkEnumerateInstanceVersionChain* pChain, uint32_t* pApiVersion) {
+    VkResult res = pChain->pfnNextLayer(pChain->pNextLink, pApiVersion);
+    *pApiVersion = VK_MAKE_API_VERSION(1, 2, 3, 4);
```

Comment:
This should be something that is configured in the TestLayer struct.

Reply:
Are you talking about the value or the exposing of the function?

Reply:
Ok.  Modified test layer so that I could set the value and compare against the set value.